gammapy/maps/axes.py

```diff
@@ -1885,6 +1945,32 @@ def __init__(self, edges_min, edges_max, reference_time, name="time", interp="li
         if np.any(delta < 0 * u.s):
             raise ValueError("Time intervals must not overlap.")
 
+    @property
+    def is_contiguous(self):
+        """Whether the axis is contiguous"""
+        return np.all(self.edges_min[1:] == self.edges_max[:-1])
+
+    @property
+    def contiguous(self):
+        """Make the time axis contiguous
+
+        Returns
+        -------
+        axis : `TimeMapAxis`
+            Contiguous time axis
+        """
+        values = np.unique(
+            [self.edges_min.to_value("s"), self.edges_max.to_value("s")]
+        )
+        edges = u.Quantity(values, unit="s")
```

Comment:
Why do we not keep the initial data unit of the Time object?

Reply:
I noticed that `np.unique([])` does not keep the  `Quantity`, however `np.unique(np.stack([]))` does:
 ```python3
 import numpy as np
from astropy import units as u

q = [1, 2, 1] * u.s
q_2 = [2, 3, 4] * u.s
print(np.unique([q, q_2]))
print(np.unique(np.stack([q, q_2])))
 ```
 I'll change to the latter solution.

Reply:
Done.